### ISSUE TITLE
postinstall python fails without reason given

### ISSUE THREAD

**Issue by tempelmann** [Homebrew/homebrew-core#31208]
I guess my system configuration (upgraded system for over 10 years) is part of the problem. The issue here is that even though the postinstall script fails and knows that, it doesn't print any information that would help finding the issue.

`brew update` does not mention any issues.

Here's the output:

		$ brew postinstall python3
		==> Postinstalling python
		==> /usr/local/Cellar/python/3.7.0/bin/python3 -s setup.py --no-user-cfg install --force --verbose --install-scripts=/usr/local/Cellar/python/3.7.0/bin --install-lib=/usr/local/lib/python3.7/site-packages --single-version-externally-managed --record=installed.txt
		Warning: The post-install step did not complete successfully
		You can try again using `brew postinstall python`

Here's the gist-log: [https://gist.github.com/tempelmann/8f07f4ae8bcd438913c38238cfe647e3](https://gist.github.com/tempelmann/8f07f4ae8bcd438913c38238cfe647e3)

So, the question is: How do I figure out what the issue is?

**Comment by DomT4** [comment]
Does running the `postinstall` command with the `-dv` flag kick out any more detail?

**Comment by tempelmann** [comment]
It does:

`==> Permission denied @ rb_sysopen - /Users/tempi/Library/Logs/Homebrew/python/post_install.01.python3`

So, as it turn out, my older brew installation from 2016, which I recently tried to completely erase, had been installed with "sudo" (which was a mistake I only later realized). Clearly, my cleanup did not include this log folder outside the obvious directory where homebrew gets installed.

Fixing the permissions on this logs folder made the post-install work.

Thanks for the fast help!

So, this already leads me to these suggestions for brew:

1. The troubleshooting page at https://docs.brew.sh/Troubleshooting should perhaps mention this -dv flag. Nothing so far (and I googled for quite a while on this issue) indicated the existence of this flag.
2. This permissions error should have been printed out without the -dv option.
